src/swap_amm.py

```diff
@@ -322,14 +324,18 @@ def __init__(self, symbol1, symbol2, addresses, myswap_id):
     async def get_balance(self):
         if self.myswap_id is not None:
             myswap_pool = await src.blockchain_call.get_myswap_pool(self.myswap_id)
-        for token in self.tokens:
-            balance = 0
-            for address in self.addresses:
-                balance += await src.blockchain_call.balance_of(token.address, address)
-            if self.myswap_id is not None:
-                balance += myswap_pool[token.symbol.upper()]
-            token.balance_base = balance
-            token.balance_converted = decimal.Decimal(balance) / token.decimal_factor
+        for amm, addresses in self.addresses_dict.items():
```

Comment:
`addresses` is always a single address, right?
```suggestion
        for amm, address in self.addresses_dict.items():
```